### PR TITLE
Updated cdk script file to generate projenrc with all common parameters

### PR DESCRIPTION
**Pull request by sangramPeerI**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by SuperMohit** [comment]
@sangramPeerI 
We would also need the ability to generate a single resource. 
This won't be useful in the future as we have generated all the resources. Can you carve out this into a new function that would just take the name of the resource and generate a scaffold? cdk-import may also be optional whether we provide a schema file or not.

**Review by Zuhairahmed — APPROVED** [review]
LGTM

**Review by AndreiMatei2206 — APPROVED** [review]
LGTM

File: cdk/cdk.sh [hunk 1]
```diff
@@ -1,6 +1,20 @@
-for dir in ../cfn-resources/*;
+# This shell script can be use to generate one or multiple L1 Constrcuts from CFN resources.
+# How to execute:
+#   For all resources without arguments  :  ./cdk.sh
+#   For specific resource with arguments :  ./cdk/sh "auditing cluster"
```
Comment (maastha) [inline cdk/cdk.sh:4]: I assume this should be just `./cdk.sh "auditing"` OR `./cdk.sh "cluster"`? Other than this LGTM

**Comment by maastha** [comment]
There is also an `automating.sh` script. Can we please add comments in that as to what that script is expected to do and clarify when that is used in the process?

**Comment by maastha** [comment]
@sangramPeerI Also, would be helpful if we can include publishing related scripts as well (or could that be an issue due to NPM token, I guess, in which case ignore this comment) 

Please feel free to include/add any other scripts/commands you might have used in the process starting from generating resources till publishing resources etc. 
Idea is to try to document the process via code as much as possible so that when we run these scripts the end result (generated resource) is same as currently existing ones for consistency, ideally.

File: cdk/cdk.sh (hunk 1 above)
Comment (SuperMohit) [reply cdk/cdk.sh:4]: @sangramPeerI can you resolve this? Then I can merge

**Comment by SuperMohit** [comment]
> 

publishing script is available.

**Comment by SuperMohit** [comment]
@sangramPeerI if a resource already exists it should not run - npx projen create again.

**Comment by Zuhairahmed** [comment]
can we also resolve branch conflicts before merging?

**Comment by sangramPeerI** [comment]
@SuperMohit.  resource already exists check is working.

@Zuhairahmed  conflicts has been resolved.

**Comment by sangramPeerI** [comment]
@maastha ,
"automating.sh" script is not required anymore. We have added all changes there in cdk.sh file.
I will remove this file.

**Review by andreaangiolillo — COMMENTED** [review]
Can you get the latest changes from master as we added new github actions?

**Comment by maastha** [comment]
@sangramPeerI  Do we still need this PR? Some changes were made as part of https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/303 as well.

**Comment by Zuhairahmed** [comment]
@SuperMohit feel free to close this PR if no longer needed

**Pull request closed without merging**